metaflow/plugins/kfp/kfp.py

```diff
@@ -132,11 +132,16 @@ def __init__(
         self.notify_on_error = notify_on_error
         self.notify_on_success = notify_on_success
 
-        self._client = kfp.Client(namespace=api_namespace, userid=username, **kwargs)
+        # kfp client userid needs to have @zillowgroup.com
+        kfp_client_userid = username
+        if "@zillowgroup.com" not in kfp_client_userid:
+            kfp_client_userid += "@zillowgroup.com"
```

Comment:
given this is an open source project, let's not introduce zillow specific settings, especially hard coded.  I've noticed that this approach creates for better code anyhow.

I'm curious, why does the KFP client require it as an email with domain name and not just the alias?

Reply:
This is because the current version of the KFP authentication is pretty naive, this is fixed in KFP >= 1.2 (I think) but essentially what it does is query the Kubernetes API for all `RoleBinding` resources in the namespace on any request to the Kubeflow API service. It then looks at that list of `RoleBinding` resources and sees if any of the `subjects` match the supplied `userid`.

Given that we create the subjects in `email` format (I'm guessing because of Okta maybe?) it'll then only match and allow the authorization to pass in this case.

```bash
 ~/Code/kubeflow.v1.1   master ●  k get rolebinding -n aip-example-dev user-alexla-zillowgroup-com-clusterrole-edit -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  ...
  name: user-alexla-zillowgroup-com-clusterrole-edit
  namespace: aip-example-dev
  ...
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kubeflow-edit
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: alexla@zillowgroup.com
```

This is also why the UI periodically crashes because the query used to do this lists all `RoleBinding` resources from the entire cluster and pulls them all back into memory on every call 🤦  See [this code](https://github.com/kubeflow/kubeflow/blob/v1.1.0/components/access-management/kfam/api_default.go#L199) for more info :)

Reply:
I forget exactly how this has changed in the later versions of KFP but I'm hoping it's a lot nicer as this approach is definitely a hack as it really just sidesteps authorization as we can send through any email address 😒

Reply:
Allowing email is a good feature. Maybe we make this email suffix a variable that can be read from configs?

Reply:
I suggest we add an OS env variable and MF config named `KFP_USER_DOMAIN` that we use to construct the `kfp_client_userid` (should probably be named kfp_client_user_email?)

Much like the KFP MF configs in https://github.com/zillow/metaflow/blob/a721c60b181f9c2704ac901d9dac65da04167286/metaflow/metaflow_config.py#L57-L59

Yes, what is confusing to us is that the KF Profiles are alias only:

```
contributors:
  - name: abdulrahmans
  - name: alexla
```

Reply:
yeah in profile we use the alias in contributors and then append @zillowgroup.com later on

Reply:
yeah that is the goal of this change. we discussed this offline in detail. let me know if you still have questions.